students/nskvarch/Lesson6/mailroom4.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+# Lesson 5 exercise "the mailroom part 3", created by Niels Skvarch
+
+# Import Modules needed to run
+import sys
+import os
+
+# define global variables
+donor_db = {"Bob Johnson": [3772.32, 512.17],
+            "Fred Billyjoe": [877.33, 455.50, 23.45],
+            "Harry Richard": [1.50],
+            "Old Gregg": [1663.23, 4300.87, 10432.0],
+            "Jerry Vars": [19.95, 653.21, 99.45],
+            }
+
+menu = ("\n".join(("Welcome to the Mailroom.",
+                   "Please choose from below options:",
+                   "1 - Add a new contribution and display sample thank you letter",
+                   "2 - Display a donor Report",
+                   "3 - Create a thank you letter for all donors",
+                   "4 - Exit",
+                   "--> ")))
+
+
+# define functions
+def print_report(data_lines):
+    """Take in a table in the form of a list and display it"""
+    sizes = [0] * len(data_lines[0])
+    for line in data_lines:
+        for item_index, item in enumerate(line):
+            if len(item) > sizes[item_index]:
+                sizes[item_index] = len(item)
+
+    for line in data_lines:
+        print(line[0].ljust(sizes[0]) + "    $    " + line[1].ljust(sizes[1]) + "    " + line[2].rjust(
+            sizes[2]) + "    \t    " + line[3].rjust(sizes[3]))
+    print("\n")
+
+
+def run_report():
+    """take in the donor database, create organize and sort a table in the form of a list and then hand the table off
+    to the printer function """
+    data_lines = []
+    for name in donor_db:
+        data = donor_db[name]
```

Comment:
you can avoid this line altogether by using .items() function (the most common way to loop over dictionary.

```
for name, data in donor_db.items():
```